src/qml/qgismobileapp.qml

```diff
@@ -509,7 +509,51 @@ ApplicationWindow {
   }
 
   Column {
-    id: mainToolBar
+    id: mainToolbar
+    anchors.left: mainMenuBar.left
+    anchors.top: mainMenuBar.bottom
+    anchors.leftMargin: 4 * dp
+    spacing: 4 * dp
+
+    Button {
+      id: topologyButton
+      round: true
+      visible: stateMachine.state === "digitize" && ( dashBoard.currentLayer.geometryType() === QgsWkbTypes.PolygonGeometry || dashBoard.currentLayer.geometryType() === QgsWkbTypes.LineGeometry )
```

Comment:
As far as I understand this, it doesn't concern the selected layer but still is only visible when a layer is selected?
Shouldn't it be visible as soon as there are at least two layers with geometry line or polygon in the project - no matter if they are selected or not?

Reply:
@signedav , Matthias wants to avoid obscuring the canvas as much as possible, hence hiding when possible.

Reply:
how does it not concern the selected layer?

Reply:
First - in browsing mode - it's not visible. Then with enable digitizing mode, it's still not visible. Then I select a layer and it becomes visible. I activate it, and because it popped out by selecting the layer, I expect, that I turn it on only for this layer. But it does not concern (only) this layer. It concerns all the polygon and line layers. 

If I have e.g. several point layers. And I click on one, the + appears to digitize. I expect that this + concerns **only** this layer. There is not an option that I can select the layer after pressing + where to add a feature. This is QGIS behavior. And so I think the activation of the topological editing should be QGIS behavior as well. Means, not appearing on selecting a specific layer.

I suggest to make it visible all the time (as soon as there are polygon or line layers) or put it somewhere else (like e.g. in the menu like the measure tool).

Reply:
> I activate it, and because it popped out by selecting the layer, I expect, that I turn it on only for this layer.

I understand this assumption. But I also think it's obvious that when you change the layer and it stays the way it is it will be quite quickly visible to someone that it's not only for this layer.

For me, topological editing is an edge case. Interesting for some, a miracle for many. So it should take away users attention as little as possible.
It also will not do anything when editing a point layer, hence it doesn't make sense to show it there.

If it was for me, I'd even make it opt-in in the configuration of QField or the project to keep it hidden from the large portion of QField users that will never ever use it.

Reply:
I'm -1 to adding a setting here, that's just an extra layer of complexity that's not worth it really IMHO.

I think hiding it as much as we can, like it's done ATM, is a good balance.

Arguably, the bonus of having it show up in the narrow circumstances we have atm does provide a collateral benefit: educating users about the presence of this feature.